GOT-StudentVersion/GOT-StudentVersion/ViewController.swift

```diff
@@ -9,17 +9,194 @@
+            
+            if let scopeArrTitles = mySearchBar.scopeButtonTitles {
+                let currentIndex = mySearchBar.selectedScopeButtonIndex
+                let selectedStr = scopeArrTitles[currentIndex]
+                
+                return GOTEpisode.getFilteredResults(selectedStr: selectedStr, searchText: searchString)
+            }
+            
+            return seasons
+        }
+    }
+    
+    var searchString: String? = nil {
+        didSet {
+            self.myTableView.reloadData()
+        }
+    }
+    
+    @IBOutlet weak var myTableView: UITableView!
+    
+    @IBOutlet weak var mySearchBar: UISearchBar!
+    
     override func viewDidLoad() {
         super.viewDidLoad()
-        // Do any additional setup after loading the view, typically from a nib.
+        configureTableView()
+        configureSearchBar()
     }
 
     override func didReceiveMemoryWarning() {
         super.didReceiveMemoryWarning()
         // Dispose of any resources that can be recreated.
     }
+    
+    func configureTableView() {
+        myTableView.dataSource = self
+        myTableView.delegate = self
+        myTableView.rowHeight = 80
+        myTableView.sectionHeaderHeight = 40
+        myTableView.tableFooterView = UIView()
+    }
+    
+    func configureSearchBar() {
+        mySearchBar.delegate = self
+    }
+    
+    
+
+
+}
+
+extension ViewController: UITableViewDataSource, UITableViewDelegate {
+   
+    
+    func tableView(_ tableView: UITableView, numberOfRowsInSection section: Int) -> Int {
+        return filteredSeasons[section].count
+    }
+    
+    func numberOfSections(in tableView: UITableView) -> Int {
+        return 7
+    }
+
+    func tableView(_ tableView: UITableView, cellForRowAt indexPath: IndexPath) -> UITableViewCell {
+        
+        let episode = filteredSeasons[indexPath.section][indexPath.row]
+       
+        let image: UIImage = (UIImage(named: episode.mediumImageID ))!
+        
+        if indexPath.section % 2 == 0 {
+            if let cell = myTableView.dequeueReusableCell(withIdentifier: "SeasonTableViewCell", for: indexPath) as? SeasonTableViewCell {
+            
+                cell.nameLabel.text = episode.name
+                cell.seasonAndNumberLabel.text = "S:\(episode.season) E:\(episode.number)"
+                cell.mediumImageLabel.image = image
+                return cell
+            }
+        } else {
+            if let cell = myTableView.dequeueReusableCell(withIdentifier: "SeasonTableViewCell2", for: indexPath) as? SeasonTableViewCell2 {
+                cell.nameLabel.text = episode.name
+                cell.seasonAndNumberLabel.text = "S:\(episode.season) E:\(episode.number)"
+                cell.mediumImageLabel.image = image
+                return cell
+            }
+        }
+        let cell2 = myTableView.dequeueReusableCell(withIdentifier: "SeasonTableViewCell", for: indexPath)
```

Comment:
since you have an if else block you dont have to return anything else outside of those